### DISCUSSION TITLE
ARM Template Specs

### DISCUSSION THREAD

**Discussion by mmisztal1980** [AzureCommunityPL/group#203] [Hyde Park]
https://dev.to/azure/arm-template-specs-now-public-preview-5ap5

Wyłapałem dzisiaj takie ogłoszenie na LinkedIn, ktoś już miał z tym do czynienia? 
Jakie są potencjalne zastosowania?
Czy dobrze interpretuję że to takie "szablony do szablonów ARM" ?

**Comment by miqm** [comment]
raczej artifactory. największą zaletą jest że możesz zrobić linked template relatywnie i wrzucając do template-spec wrzuci ci oba pliki. czyli w końcu można sensownie rozbijać ARM na moduły bez konieczności ogrywania tego storage accountem i SASami.

**Reply by kamil-mrzyglod** [reply]
To + możesz jednak do speca zrobić referencję poprzez `templateLink` w swoim głównym szablonie więc jest to jakaś doza modułowości w skali organizacji. Minus tego ja widzę taki, że każdy komponent jest wtedy robiony poprzez oddzielny deployment a nie wiem jeszcze jak sprawa ma się z outputami z takich deploymentów chociaż na pewno da radę się do nich dobić :)

TF nadal robi to lepiej.

**Reply by mmisztal1980** [reply]
Zobaczymy co na to Bicep :D

**Reply by miqm** [reply]
Z Bicep widzę jeden główny problem - moduły bicepowe są jako nested template - i jako takie są widoczne - każdy osobno - na liście historii deploymentów. Tutaj dygresja - nazywając moduł w bicep ja używam np. `'${deployment().name}-www`, żeby je jakoś sensownie zgrupować.

Jednak nie jestem pewien teraz jak wygląda to z template-specs - czy jest widoczny jako jeden wpis czy jednak jako osobne - a niestety nie mam żadnego stworzonego pod ręką.
